js/common/lib/backend.ts

```diff
@@ -44,6 +56,11 @@ export interface Backend {
 
   createSessionHandler(uriOrBuffer: string|Uint8Array, options?: InferenceSession.SessionOptions):
       Promise<SessionHandler>;
+
+  createTrainingSessionHandler?
```

Comment:
you can use the type `UriOrBuffer` defined as above, then shorten the parameter name. And also make `evalModel` and `optimizerModel` optional.

Is session option also optional parameter or required? if it's also optional you can modify the parameter list to:

```ts
  createTrainingSessionHandler?(
        checkpoint: UriOrBuffer,
        trainModel: UriOrBuffer,
        evalModel?: UriOrBuffer,
        optimizerModel?: UriOrBuffer,
        sessionOptions?: InferenceSession.SessionOptions): Promise<TrainingSessionHandler>;
```

Reply:
The session options are optional, but to avoid multiple function overloads & a lot of type-narrowing for a function that the user won't ever directly use, I'm having the TrainingSessionImpl pass in default values for evalModel, optimizerModel, and sessionOptions to the handler that will be treated as optional (empty strings & empty maps).